template/pdk.gen.go.ejs

```diff
@@ -179,20 +176,17 @@ func _<%- goName(ex.name) %>() int32 {
 	<% } -%>
 	
 	<% if (imp.output) { -%>
-		<% if (imp.output.contentType === 'application/json') { -%>
+		<% if (isJsonEncoded(imp.output)) { -%>
 			var output <%- toGolangType(imp.output) %>
 			err = pdk.JSONFrom(offs, &output)
 			if err != nil {
 				return nil, err
 			}
 			return &output, nil
-		<% } else if (imp.output.contentType === 'text/plain; charset=UTF-8') { -%>
-			outputMem := pdk.FindMemory(offs)
-			return string(outputMem.ReadBytes()), nil
-		<% } else if (imp.output.type === 'string') { -%>
+		<% } else if (isString(imp.output) || isUtf8Encoded(imp.output)) { -%>
```

Comment:
I merged these two together. not sure why one was returning a string value and the other was a pointer value. Do we generically use utf encoded values for anything other than strings?

Reply:
enum maybe?

Reply:
I'll take a close look, but there was some stuff where I had to return a *T so it was easier to return the `nil` in the case where I'd encountered an `error` and wanted to return the `nil, err` pair vs. trying to figure out what kind of default I needed to construct to go along with the returned error.

(this might not be the reason, but I'll confirm soon)

Reply:
(nope, totally unrelated lol) 

I think this was just a symptom of "list all the possible cases, then code" not thinking these would have the same operation.